### ISSUE TITLE
Feature request: filter warnings to only current lib/app

### ISSUE THREAD

**Issue by jasonkarns** [rspec/rspec-core#2192]
I love that rspec helps me resolve warnings, but the typical app (even gem) has so many other gem deps emitting warnings that it's difficult to sift through the noise for the signal.

I would love the ability for rspec's warnings to be filtered to only show those emanating from the current working directory.

**Comment by myronmarston** [comment]
There was a recent discussion on the mailing list about this:

https://groups.google.com/forum/#!topic/rspec/ua4K3Rdv3uU

The TL;DR is that it's a great idea but, as someone who has tried to implement this previously, it's extremely difficult to implement in a simple, robust way, because Ruby provides no API for controlling warnings.  Your only interface to warnings is the Ruby flag to enable them and `$stderr` and warnings don't necessarily tell you where they come from.  Plus `$stderr` has lots of other output that's not directly related to Ruby warnings (such as deprecation warnings from libraries, any code that calls `Kernel#warn`, etc).

IMO, the best path forward is to work with Ruby core to get some APIs added for controlling warnings in a granular way.

**Comment by jasonkarns** [comment]
ah, thanks!
